src/channel_internal.c

```diff
@@ -392,10 +481,10 @@ IMPLEMENT_MOCKABLE_FUNCTION(, CHANNEL_RESULT, channel_internal_push, THANDLE(CHA
         /*Codes_SRS_CHANNEL_INTERNAL_43_124: [ If the first operation in the list of pending operations contains a non-NULL pull_callback: ]*/
         else if (CONTAINING_RECORD(channel_internal_ptr->op_list.Flink, CHANNEL_OP, anchor)->pull_callback != NULL)
         {
-            if (dequeue_operation(channel_internal, out_op_push, NULL, NULL, push_callback, push_context, data) != 0)
+            if (dequeue_operation(channel_internal, out_op_push, NULL, NULL, NULL, correlation_id, push_callback, push_context, data) != 0)
             {
                 /*Codes_SRS_CHANNEL_INTERNAL_43_041: [ If there are any failures, channel_internal_push shall fail and return CHANNEL_RESULT_ERROR. ]*/
-                LogError("Failure in dequeue_operation(channel_internal_ptr, out_op_push, NULL, NULL, push_callback, push_context, data)");
+                LogError("Failure in dequeue_operation(channel_internal_ptr, out_op_push, NULL, NULL, NULL, push_callback, push_context, data)");
```

Comment:
>ailure in dequeue_operation(channel_internal_ptr, out_op_push, NULL, NULL, NULL, push_callback, push_context, data)"); [](http://example.com/codeflow?start=27&length=118)

should log all arguments with their values.

(including very important "correlation_id" I would say.

Reply:
[https://msazure.visualstudio.com/One/_backlogs/backlog/Azure%20Messaging%20GeoReplication/Backlog%20items?workitem=29879341](https://msazure.visualstudio.com/One/_backlogs/backlog/Azure%20Messaging%20GeoReplication/Backlog%20items?workitem=29879341)